src/backend/common/helpers/matchstats_helper.py

```diff
@@ -0,0 +1,72 @@
+from typing import Dict, Callable, List
+
+from backend.common.consts.alliance_color import AllianceColor
+from backend.common.models.match import Match
+from backend.common.helpers.opr_helper import OPRHelper
+
+
+def get_opp_color(color: AllianceColor) -> AllianceColor:
+    return [AllianceColor.RED, AllianceColor.BLUE][color == AllianceColor.RED]
+
+
+class MatchstatsHelper:
+    EVERGREEN: Dict[str, Callable[[Match, AllianceColor], float]] = {
+        "oprs": lambda match, color: match.alliances[color]["score"],
+        "dprs": lambda match, color: match.alliances[get_opp_color(color)]["score"],
+        "ccwms": lambda match, color: (
+            match.alliances[color]["score"]
+            - match.alliances[get_opp_color(color)]["score"]
+        ),
+    }
+
+    COMPONENTS = {
+        2020: {
+            "Foul Points": lambda match, color: match.score_breakdown[color].get(
+                "foulPoints", 0
+            ),
+            "Auton div Total": lambda match, color: (
+                match.score_breakdown[color].get("autoPoints", 0)
+                / max(match.score_breakdown[color].get("totalPoints", 0), 1)
+            ),
+        }
+    }
+
+    component_default = lambda comp: (
+        lambda match, color: match.score_breakdown[color].get(comp, 0)
+    )
+
+    @classmethod
+    def calculate_matchstats(cls, matches: List[Match]):

```

Comment:
annotate a return type, you can do `models.stats.EventMatchStats`, I think (assuming we add the right stuff to the stat type enum)